### PR TITLE
[chore] OSS: add a small sphinx tutorial, similar to README

### PR DESCRIPTION
**Pull request by blefaudeux**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Adds a tutorial for OSS in the html docs. cf #63 

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/tutorials/oss.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        # DDP
+        dist_init(rank, world_size)
+
+        # Problem statement
+        model = myAwesomeModel()
+        dataloader = mySuperFastDataloader()
+        loss = myVeryRelevantLoss()
+
+        base_optimizer_arguments = {} # pass any optimizer specific arguments here, or directly below when instantiating OSS
+        optimizer = torch.optim.SGD(params=model.parameters() **base_optimizer_arguments)
```
Comment (stephenroller) [inline docs/source/tutorials/oss.rst:26]: typo?

Comment (blefaudeux) [reply docs/source/tutorials/oss.rst:26]: quick copy paste, not the nicest, I meant to minimize the changes in between the two configs. You mean about the missing comma (fixing that :/) or worse, that it's not very clear ?

File: fairscale/optim/oss.py [hunk 2]
```diff
@@ -46,7 +46,9 @@ class OSS(Optimizer):
             torch.distributed group (default: group.WORLD)
     """
 
+    #: The optimizer used for a given shard
```
Comment (min-xu-ai) [inline fairscale/optim/oss.py:49]: just curious, is this : and extra space below significant for doc purpose?

**Review by min-xu-ai — APPROVED** [review]
LGTM

File: fairscale/optim/oss.py (hunk 2 above)
Comment (blefaudeux) [reply fairscale/optim/oss.py:49]: ah no, I think that was a reformat from the IDE, it changes depending on the black version, it's pretty annoying

**Pull request merged**